src/lidar_mini.cpp

```diff
@@ -21,30 +21,35 @@ std::array<char, 9> LIDARmini::getWantedRegisters(char registerSetByte) {
     // But we think it might be related to the BAUDrate
     for (int j = 0; j < 100000; j++) {
         char c;
+        while (!uart.available()) {
+        }
         uart >> c;
 
         if (c == 0x59) {
             char dat[8];
-
             for (int i = 0; i < 8; i++) {
+                while (!uart.available()) {
+                }
                 uart >> dat[i];
             }
-
+            for (int i = 0; i < 8; i++) {
+                hwlib::cout << hwlib::hex << (int)dat[i] << "\r\n";
```

Comment:
You don't need the \r, just fix your line endings in PuTTY (I assume that is what you use).